cpe/attributes/java/java.go

```diff
@@ -90,6 +120,15 @@ func getPossibleVendors(origins []string) set.StringSet {
 	if vendorSet.Cardinality() == 0 {
 		vendorSet.Add("apache")
 	}
+
+	// Add Spring-specific vendors.
```

Comment:
Does it make sense to do this only _after_ adding `"apache"` based on nothing but hope? Isn't the `knownSpringComponents` check more precise than the `"apache"` default and therefore should suppress the latter?

Reply:
I guess another question would be, is it adequate to do this here even when `vendorSet` is non-empty (and wasn't populated via the apache fallback)?